core/src/worker/request-response-L2-request.ts

```diff
@@ -0,0 +1,115 @@
+import { L1Endpoint__factory } from '@bisonai/orakl-contracts'
+import { Queue, Worker } from 'bullmq'
+import { ethers } from 'ethers'
+import { Logger } from 'pino'
+import type { RedisClientType } from 'redis'
+import { getVrfConfig } from '../api'
+import {
+  BULLMQ_CONNECTION,
+  CHAIN,
+  L1_ENDPOINT,
+  L2_REPORTER_REQUEST_RESPONSE_REQUEST_QUEUE_NAME,
+  L2_WORKER_REQUEST_RESPONSE_REQUEST_QUEUE_NAME,
+  REQUEST_RESPONSE_FULFILL_GAS_MINIMUM,
+  WORKER_JOB_SETTINGS
+} from '../settings'
+import {
+  IL2RequestResponseListenerWorker,
+  IL2RequestResponseRequestTransactionParameters,
+  ITransactionParameters,
+  IVrfConfig,
+  QueueType
+} from '../types'
+
+const FILE_NAME = import.meta.url
+
+export async function worker(redisClient: RedisClientType, _logger: Logger) {
+  const logger = _logger.child({ name: 'worker', file: FILE_NAME })
+  const queue = new Queue(L2_REPORTER_REQUEST_RESPONSE_REQUEST_QUEUE_NAME, BULLMQ_CONNECTION)
+  // FIXME add checks if exists and if includes all information
+  const vrfConfig = await getVrfConfig({ chain: CHAIN, logger })
+  const worker = new Worker(
+    L2_WORKER_REQUEST_RESPONSE_REQUEST_QUEUE_NAME,
+    await job(queue, vrfConfig, _logger),
+    BULLMQ_CONNECTION
+  )
+
+  async function handleExit() {
+    logger.info('Exiting. Wait for graceful shutdown.')
+
+    await redisClient.quit()
+    await worker.close()
+  }
+  process.on('SIGINT', handleExit)
+  process.on('SIGTERM', handleExit)
+}
+
+export async function job(reporterQueue: QueueType, config: IVrfConfig, _logger: Logger) {
```

Comment:
`config` parameter is not used.

```suggestion
export async function job(reporterQueue: QueueType, _logger: Logger) {
```